Character.cpp

```diff
@@ -88,30 +85,40 @@ int Character::takeDamage(int damage)
 
 
 #include <cassert>
-void Character::attackInternal(Character& other)
+void Character::attackInternal (Character& other)
 {
-    if( other.hitPoints <= 0 )
+    if ( other.hitPoints <= 0 )
     {
         /*
         When you defeat another Character: 
             a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if (hitPoints < *initialHitPoints.get()) hitPoints = *initialHitPoints;
+        if (armor < *initialArmorLevel.get()) armor = *initialArmorLevel;
+        if (attackDamage < *initialAttackDamage.get()) attackDamage = *initialAttackDamage;
+
+        hitPoints *= 1.1f;
+        armor *= 1.1f;
+        attackDamage *= 1.1f;
+
+        initialHitPoints.reset (new int (hitPoints));
+        initialArmorLevel.reset (new int (armor));
+        initialAttackDamage.reset (new int (attackDamage));
```

Comment:
You're doing the same thing 3 times in a row. 
D.R.Y. refactor this into a function and call it 3x. 

Also, don't `reset(new` an integer.   `*pointer` gives you access to the underlying variable being pointed to, and you can read from it as well as write to it.  `reset(new` is an unnecessary heap allocation.
- [ ] revise